tests/src/Functional/DeveloperTest.php

```diff
@@ -301,4 +351,41 @@ protected function developerCreateByAdminTest() {
     $this->assertFalse($this->developerStorage->loadUnchanged($test_user['email']), 'Developer does not exists anymore.');
   }
 
+  /**
+   * Tests getCompanies() function of the developer entity.
+   *
+   * @see \Drupal\apigee_edge\Entity\Developer::getCompanies()
+   */
+  public function developerGetCompanyListTest() {
+    // Result of getCompanies() function cannot be null.
+    $this->assertNotNull($this->developer->getCompanies());
+
+    // Create a new company and add developer as a member to it.
+    $this->company = new Company([
+      'name' => $this->getRandomGenerator()->name(),
+    ]);
+    $company_controller = new CompanyController($this->sdkConnector->getOrganization(), $this->sdkConnector->getClient());
+    $company_controller->create($this->company);
+
+    $company_membership_controller = new CompanyMembersController($this->company->getName(), $this->sdkConnector->getOrganization(), $this->sdkConnector->getClient());
+    $company_membership = new CompanyMembership([$this->developer->getEmail() => NULL]);
+    $company_membership_controller->setMembers($company_membership);
+
+    // Ensure that the developer is loaded from Apigee Edge so remove the
+    // developer entity from the cache.
+    $developer_cache = $this->container->get('apigee_edge.controller.cache.developer');
+    $developer_cache->removeEntities([$this->developer->getDeveloperId()]);
+
+    // Check the companies array if the developer is loaded by loadMultiple().
+    /** @var \Drupal\apigee_edge\Entity\DeveloperInterface $developer */
+    $developer = Developer::loadMultiple()[$this->developer->getEmail()];
+    $this->assertContains($this->company->getName(), $developer->getCompanies());
+
+    $developer_cache->removeEntities([$this->developer->getDeveloperId()]);
+
+    // Check the companies array if the developer is loaded by load().
+    $developer = Developer::load($this->developer->getEmail());
+    $this->assertContains($this->company->getName(), $developer->getCompanies());
+  }
```

Comment:
Missing test scenario:
* Remove the developer from the company.
* Reset the entity controller's static cache.
* getCompanies() should not return that the developer is a member in the company.